Assets/Scripts/Framework/Statemachine/SmallEnmiesBaseState.cs

```diff
@@ -0,0 +1,26 @@
+public abstract class SmallEnmiesBaseState : BaseState
+{
+    #region BaseState to SmallEnmiesState
+
+    public override void EnterState(StateMachine entity)
+    {
+        base.EnterState(Parent);
+        EnterState((SmallEnemiesStateMachine) Parent);
```

Comment:
I do wonder though, why is a Parent something that is constantly being used as a parameter for methods such as "EnterState"?